### PR TITLE
Add Wave Info To Fight Cave Plugin

### PR DESCRIPTION
**Pull request by OTRD5k**
The fight cave plugin now shows what type of enemies to expect, and how many to expect for that wave.

See bottom right corner

image

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 1]
```diff
@@ -53,6 +56,74 @@
 
 	private JadAttack attack;
 
+	private static final ArrayList<Integer> FIGHT_CAVE_REGIONS = new ArrayList<Integer>(Arrays.asList(9294, 9295, 9296, 9550, 9551, 9552, 9806, 9807, 9808));
+
+	private static final int[][] WAVE_ENEMIES = {//I could have just written a function, but this is faster to write :shrug:
+			{22},
```
Comment (LeviSchuck) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: I think a lookup table like this is fine as is.
I recommend you add a comment before this saying each wave is a list of monsters that are present in that wave. That would be a more helpful comment.

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java [hunk 2]
```diff
@@ -65,17 +70,70 @@ public Dimension render(Graphics2D graphics)
...
+					else
+					{
+						numberOfEnemies.put(enemies[i], 1);
+					}
+				}
+				PanelComponent waveInfo = new PanelComponent();
+				waveInfo.setTitle("Current Wave: " + Integer.toString(wave));
+				List<PanelComponent.Line> lines = waveInfo.getLines();
+
+				if (numberOfEnemies.containsKey(360))
```
Comment (LeviSchuck) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java]: Could have made a lookup table here too, but this is still fine.

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 3]
```diff
@@ -88,13 +159,59 @@ else if (jad.getAnimation() == JadAttack.RANGE.getAnimation())
 		}
 	}
 
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() == ChatMessageType.SERVER && isInFightCaves())
+		{
+			String msg = Text.removeTags(event.getMessage());//.substring(6);
```
Comment (LeviSchuck) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: Remove commented code

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 4]
```diff
@@ -88,13 +159,59 @@ else if (jad.getAnimation() == JadAttack.RANGE.getAnimation())
...
+			if (msg.contains("Wave:"))
+			{
+				this.wave = Integer.parseInt(msg.substring(6));
+				return;
+			}
+		}
+	}
+
+	private boolean isInFightCaves()//more like is in city or fight caves, but close enough
```
Comment (LeviSchuck) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: Please move comment to before the function

**Comment by jpabeem** [comment]
My PR for this functionality (#1896) was shutdown, haven't found the time yet to improve it. Maybe you can do something with the feedback.

**Comment by OTRD5k** [comment]
Fixed all the commenting problems

**Comment by permedd** [comment]
yes! i want this!

**Pull request closed without merging**